hdrline.c

```diff
@@ -497,7 +498,7 @@ static const char *index_format_str(char *buf, size_t buflen, size_t col, int co
 #endif
```

Comment:
Good catch - do we perhaps want to rewrite it as follows?
```c
if (m)
{
  p = NULL;
#ifdef USE_NOTMUCH
  if (m->type == MUTT_NOTMUCH)
  {
    p = nm_email_get_folder_rel_db(m, e);
  }
#endif
  if (!p)
  {
    p = strrchr(mailbox_path(m), '/');
    if (p)
    {
      p++;
    }
  }
  if (p)
  {
    mutt_str_copy(buf, p, buflen);
  }
}
```

Reply:
Sure, I've updated the PR to match.